Add tests for the _jit_wrapper entry point

steven-johnson opened the pull request:
Add to example_aottest and user_context_aottest. (Issue #667)

abadams commented:
I'm a little wary about changing the example aot test, because it's the minimal example we can point to for how to call into Halide as produced by a generator. Do you think it's worth moving this check to its own test?

steven-johnson commented:
will do

steven-johnson commented:
BTW: while I'm here, do we want to entertain giving it a less-weirdly-confusing name?

steven-johnson commented:
renamed _jit_wrapper -> _argv (per Dillon's suggestion); added separate testcase and reverted the change to "example"; added prototype to generated header files. PTAL

abadams commented on test/generator/argvcall_aottest.cpp:
```diff
@@ -0,0 +1,53 @@
+  // We can, of course, pass whatever values for Param/ImageParam that we like.
+  result = argvcall(1.2f, 3.4f, output);
+  if (result != 0) {
+    fprintf(stderr, "Result: %d\n", result);
```
Wrong indentation in this function

abadams commented:
lgtm apart from the indentation nit
